CMakeLists.txt

```diff
@@ -177,9 +177,13 @@ endif()
 
 # uClibc *intentionally* crashes in duplocale(), at least as of:
 # https://github.com/ffainelli/uClibc/blob/266bdc1/libc/misc/locale/locale.c#L1322
-# So, if it looks like we're compiling for a system like that just disable 
+# So, if it looks like we're compiling for a system like that just disable
 # locale handling entirely.
-exec_program(${CMAKE_C_COMPILER} ARGS -dumpmachine OUTPUT_VARIABLE CMAKE_GNU_C_MACHINE)
+if (CMAKE_VERSION GREATER_EQUAL "3.28")
```

Comment:
The execute_process function exists with all the flags you're using in at least the minimum version we require (https://cmake.org/cmake/help/v3.9/command/execute_process.html), so it seems unnecessary to make this conditional.

Reply:
Yeah, but warning emitted only on v3.28+, so that's maybe better than.